src/shared/js/request.html

```diff
@@ -106,21 +110,21 @@
 					instance:this,
 					response:null
 				};
-				this.promise.reject(detail);
-				this.dispatchEvent(new CustomEvent(_eventPrefix+"preflight"), detail);
+				this.defer.reject(detail);
+				this.fire('preflight',detail);
 			}
 		},
 
 		handleProgress: function(e) {
 			var detail = {
-					percent: e.totalSize/e.position,
-					total: e.totalSize,
-					current: e.position
+					percent: e.total || e.totalSize/e.loaded || e.position,
```

Comment:
Watch out order of operations here (parens around the `||`ed or subexpressions) and make sure you want `total/loaded` and not `loaded/total`